src/renate/updaters/avalanche/model_updater.py

```diff
@@ -0,0 +1,328 @@
+            plugins=plugins,
+            evaluator=evaluator,
+            device=self._get_device(),
+            eval_every=1,
+        )
+        avalanche_learner.is_logged_metric = (
+            lambda metric_name: metric_name is None or metric_name in metrics_mapper.keys()
+        )
+        return avalanche_learner
+
+    @staticmethod
+    def _load_if_exists(
+        current_state_folder: Optional[str], metric: str, mode: defaults.SUPPORTED_TUNING_MODE_TYPE
+    ) -> Optional[Naive]:
+        """Loads the Avalanche strategy if a state exists."""
+
+        if current_state_folder is None:
+            return None
+        checkpoint_plugin = RenateCheckpointPlugin(
+            RenateFileSystemCheckpointStorage(
+                directory=Path(current_state_folder),
+            ),
+            metric=metric,
+            mode=mode,
+            # map_location=self._devices TODO
+        )
+        avalanche_learner, _ = checkpoint_plugin.load_checkpoint_if_exists()
+        return avalanche_learner
+
+    def update(
+        self,
+        train_dataset: Dataset,
+        val_dataset: Optional[Dataset] = None,
+        task_id: Optional[str] = None,
+    ) -> RenateModule:
+        if isinstance(train_dataset, _TransformedDataset):
+            x_data, y_data = [], []
+            for x, y in train_dataset:
+                x_data.append(x)
+                y_data.append(y)
+            train_dataset = TensorDataset(torch.stack(x_data), torch.stack(y_data))
+        val_memory_dataset = train_dataset
+        val_dataset_exists = val_dataset is not None
+        if val_dataset_exists:
+            self._dummy_learner._val_memory_buffer.update(val_dataset)
+            val_memory_dataset = TensorDataset(*self._dummy_learner._val_memory_buffer.to_tensors())
+        self._learner.evaluator.loggers[0].report_val = val_dataset_exists
+        benchmark = dataset_benchmark(
+            [train_dataset],
+            [val_memory_dataset],
+            train_transform=self._train_transform,
+            train_target_transform=self._train_target_transform,
+            eval_transform=self._test_transform,
+            eval_target_transform=self._test_target_transform,
+        )
+        train_exp = benchmark.train_stream[0]
+        self._learner.train(train_exp, eval_streams=[benchmark.test_stream])
+        torch.save(self._model.state_dict(), defaults.model_file(self._next_state_folder))
+        if val_dataset_exists:
+            torch.save(
+                self._dummy_learner._val_memory_buffer.state_dict(),
+                Path(self._next_state_folder) / defaults.BUFFER_CHECKPOINT_NAME,
+            )
+        results = self._learner.eval(benchmark.test_stream)
+        SyneTunePlugin.report(
+            epoch=self._max_epochs, results=results, report_val=val_dataset_exists
+        )
+
+        return self._model
+
+
+class ExperienceReplayAvalancheModelUpdater(AvalancheModelUpdater):
```

Comment:
I know this follows the same structure of the other model updaters but since in this case we will always have a fairly standard model updater instantiating a specific learner I wonder if we should not consider having an AvalancheModelUpdater -- which we already have -- that instantiates a learner based on an input parameter (kwargs for the extra arguments).
We can discuss this offline.